virtio-devices/src/vsock/unix/muxer.rs

```diff
@@ -480,55 +496,73 @@ impl VsockMuxer {
         }
     }
 
+    fn parse_port_from_read_command(command: &PartiallyReadCommand) -> Result<u32> {
+        // both lower and upper case connect statement are supported
+        const CONNECT_WITH_SPACE_LOWER_CASE: &str = "connect ";
+        const CONNECT_WITH_SPACE_UPPER_CASE: &str = "CONNECT ";
+        let opt_new_line_position = command.buf[..command.len].iter().position(|x| *x == b'\n');
+
+        // we need to read more to get a `connect ` statement
+        if command.len < CONNECT_WITH_SPACE_LOWER_CASE.len() {
+            return match opt_new_line_position {
+                Some(_) => Err(Error::InvalidPortRequest),
+                None => Err(Error::UnixRead(std::io::ErrorKind::WouldBlock.into())),
+            };
+        }
+
+        // check for both upper and lower case connect statements
+        if !command
+            .buf
+            .starts_with(CONNECT_WITH_SPACE_LOWER_CASE.as_bytes())
+            && !command
+                .buf
+                .starts_with(CONNECT_WITH_SPACE_UPPER_CASE.as_bytes())
```

Comment:
I find these multiple constants quite messy, especially since you have to get the length of an arbitrary one (or mess around with max).  Something like `command[.."connect ".len()].eq_ignore_ascii_case("connect ")` would be cleaner.